src/validation.h

```diff
@@ -199,22 +199,33 @@ static const uint64_t MIN_DISK_SPACE_FOR_BLOCK_FILES = 550 * 1024 * 1024;
  * block is made active. Note that it does not, however, guarantee that the
  * specific block passed to it has been checked for validity!
  *
- * If you want to *possibly* get feedback on whether pblock is valid, you must
- * install a CValidationInterface (see validationinterface.h) - this will have
- * its BlockChecked method called whenever *any* block completes validation.
+ * Performs initial sanity checks using the provided BlockValidationState before
+ * connecting any block(s). If you want to *possibly* get feedback on whether
+ * pblock is valid beyond just cursory mutation/DoS checks, you must install
+ * a CValidationInterface (see validationinterface.h) - this will have its
+ * BlockChecked method called whenever *any* block completes validation (note
```

Comment:
Perhaps I'm understanding wrong but this shouldn't apply anymore?

Reply:
BlockChecked is called for any block that we try to connect to the chain. See the call in `ConnectTip()`:

- if the call to `ConnectBlock()` failed, then we'll call `BlockChecked()` with a state that is NOT `IsValid()`.
- if the call to `ConnectBlock()` succeeded, then the block has been connected and is part of the main chain, and we'll call `BlockChecked()` with a state that IS `IsValid()`.

Note that we may never even call `ConnectTip()` if we don't try to connect the block to the most work chain.